client/src/components/ContentNavigator/convert.ts

```diff
@@ -0,0 +1,198 @@
+// Copyright © 2023, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+const stepRef: Record<string, string> = {
+  sas: "a7190700-f59c-4a94-afe2-214ce639fcde",
+  sql: "a7190700-f59c-4a94-afe2-214ce639fcde",
+  python: "ab59f8c4-af9a-4608-a5d5-a8365357bb99",
+};
+
+const stepTitle: Record<string, string> = {
+  sas: "SAS Program",
+  sql: "SQL Program",
+  python: "Python Program",
+};
+
+interface Entry {
+  language: string;
+  code: string;
+}
+
+function getPropPort(idx: number, inputList: Entry[]): Record<string, string> {
+  if (idx === 0) {
+    return {
+      "UI_PROP_PORT_DESCRIPTION|outTables|0": "Output tables",
+      "UI_PROP_PORT_LABEL|outTables|0": "Output table 1",
+    };
+  } else if (idx === inputList.length - 1) {
+    return {
+      "UI_PROP_PORT_DESCRIPTION|inTables|0": "Input tables",
+      "UI_PROP_PORT_LABEL|inTables|0": "Input table 1",
+    };
+  } else {
+    return {
+      "UI_PROP_PORT_DESCRIPTION|inTables|0": "Input tables",
+      "UI_PROP_PORT_LABEL|inTables|0": "Input table 1",
+      "UI_PROP_PORT_DESCRIPTION|outTables|0": "Output tables",
+      "UI_PROP_PORT_LABEL|outTables|0": "Output table 1",
+    };
+  }
+}
+
+function generateFlowData(inputList: Entry[], outputFile: string) {
+  const now = new Date();
+  const nowString = now.toISOString();
+  const nowTimestamp = String(now.getTime());
+  const arrayIdNode: string[] = [];
+  const flowData = {
+    creationTimeStamp: nowString,
+    modifiedTimeStamp: nowString,
+    createdBy: "user@sas.com",
+    modifiedBy: "user@sas.com",
+    version: 2,
+    id: null,
+    name: outputFile,
+    description: null,
+    properties: {
+      UI_PROP_DF_OPTIMIZE: "false",
+      UI_PROP_DF_ID: "120081fc-2d7f-4cfc-bcd3-47f9684e9763",
+      UI_PROP_DF_EXECUTION_ORDERED: "false",
+    },
+    links: [],
+    nodes: {},
+    parameters: {},
+    connections: [],
+    extendedProperties: {},
+    stickyNotes: [],
+  };
+
+  for (let idx = 0; idx < inputList.length; idx++) {
+    const idNode = `id-${Date.now()}-${Math.floor(Math.random() * 10000)}`;
+    arrayIdNode.push(idNode);
+    const idNote = `id-${Date.now()}-${Math.floor(Math.random() * 10000)}`;
+    const entry = inputList[idx];
+    const propports = getPropPort(idx, inputList);
```

Comment:
```suggestion
    const propPorts = getPropPort(idx, inputList);
```

Reply:
changed variable name in cf89c943731e7c2fef458becd0ea636475db0f59